lib/text/sbv_text_parser.js

```diff
@@ -0,0 +1,106 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.text.SbvTextParser');
+
+goog.require('goog.asserts');
+goog.require('shaka.text.TextEngine');
+goog.require('shaka.text.VttTextParser');
+goog.require('shaka.util.BufferUtils');
+goog.require('shaka.util.StringUtils');
+
+
+/**
+ * @implements {shaka.extern.TextParser}
+ * @export
+ */
+shaka.text.SbvTextParser = class {
+  /** */
+  constructor() {
+    /**
+     * @type {!shaka.extern.TextParser}
+     * @private
+     */
+    this.parser_ = new shaka.text.VttTextParser();
+  }
+
+  /**
+   * @override
+   * @export
+   */
+  parseInit(data) {
+    goog.asserts.assert(false, 'SubViewer does not have init segments');
+  }
+
+  /**
+   * @override
+   * @export
+   */
+  parseMedia(data, time) {
+    const SbvTextParser = shaka.text.SbvTextParser;
+    const BufferUtils = shaka.util.BufferUtils;
+    const StringUtils = shaka.util.StringUtils;
+
+    // Get the input as a string.
+    const str = StringUtils.fromUTF8(data);
+
+    const vvtText = SbvTextParser.sbv2webvtt(str);
+
+    const newData = BufferUtils.toUint8(StringUtils.toUTF8(vvtText));
+
+    return this.parser_.parseMedia(newData, time);
+  }
+
+  /**
+   * Convert a SRT format to WebVTT
+   *
+   * @param {!string} data
+   * @return {!string}
+   * @export
+   */
+  static sbv2webvtt(data) {
+    const SbvTextParser = shaka.text.SbvTextParser;
+    let result = 'WEBVTT\n\n';
+
+    // Supports no cues
+    if (data == '') {
+      return result;
+    }
+
+    // remove dos newlines
+    let srt = data.replace(/\r+/g, '');
+    // trim white space start and end
+    srt = srt.trim();
+
+    // get cues
+    const cuelist = srt.split('\n\n');
```

Comment:
According to https://wiki.videolan.org/SubViewer/, there can be header info too and you'd need to convert `[br]` into linebreaks.

Reply:
This format is used mainly on YouTube and as far as I know it is a subset. What is intended to implement is the support for this, in the description there is a link and an example of this.